### PR TITLE
Vault PNL cut off

### PR DESCRIPTION
**Pull request by ruixhuang**

### CODE REVIEW

File: src/commonMain/kotlin/exchange.dydx.abacus/functional/vault/Vault.kt [hunk 1]
```diff
@@ -114,7 +114,7 @@ object VaultCalculator {
             }
         }.filter { entry -> entry.date != null && entry.date >= dataCutoffMs }
 
-        val latestEntry = history.first()
```
Comment (ruixhuang) [inline src/commonMain/kotlin/exchange.dydx.abacus/functional/vault/Vault.kt:117]: first() will throw exception when list is empty.

Comment (tyleroooo) [reply src/commonMain/kotlin/exchange.dydx.abacus/functional/vault/Vault.kt:117]: oh very good call

File: src/commonTest/kotlin/exchange.dydx.abacus/tests/payloads/AbacusMockData.kt [hunk 2]
```diff
@@ -60,6 +60,7 @@ class AbacusMockData {
         chainName = "dYdX Chain",
         chainLogo = "dYdX-logo.png",
         rewardsHistoryStartDateMs = "1704844800000",
+        megavaultHistoryStartDateMs = 1704844800000.0,
```
Comment (tyleroooo) [inline src/commonTest/kotlin/exchange.dydx.abacus/tests/payloads/AbacusMockData.kt:63]: it's a string in the env.json I think

File: src/commonMain/kotlin/exchange.dydx.abacus/state/manager/Environment.kt [hunk 3]
```diff
@@ -430,6 +430,7 @@ class V4Environment(
     val chainName: String?,
     val chainLogo: String?,
     rewardsHistoryStartDateMs: String,
+    val megavaultHistoryStartDateMs: Double?,
```
Comment (tyleroooo) [inline src/commonMain/kotlin/exchange.dydx.abacus/state/manager/Environment.kt:433]: yeah it's a string

Comment (ruixhuang) [reply src/commonMain/kotlin/exchange.dydx.abacus/state/manager/Environment.kt:433]: We will parse it as double.. since that's how it's going to be used.

**Review by jaredvu — APPROVED** [review]

**Pull request merged**